### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by martinetd**
Lots of small cleanups:
 - rename init to create for some init functions that did allocs for us
 - more memory leaks, mainly examples/compositor not calling destroy functions.
 - the biggest part is probably wlr_surface, I've taken the liberty to reorder things a bit. There should not be any functional change (except that we no longer union with an empty buffer needlessly), but can be worth looking at.

### CODE REVIEW

File: backend/wayland/output.c [hunk 1]
```diff
@@ -224,7 +224,6 @@ struct wlr_output *wlr_wl_output_create(struct wlr_backend *_backend) {
 
 	wlr_output->width = 640;
 	wlr_output->height = 480;
-	wlr_output->scale = 1;
```
Comment (SirCmpwn) [inline backend/wayland/output.c]: Why did you remove this?

File: types/wlr_surface.c [hunk 2]
```diff
@@ -67,6 +67,9 @@ static void surface_set_opaque_region(struct wl_client *client,
 		struct wl_resource *resource,
 		struct wl_resource *region_resource) {
 	struct wlr_surface *surface = wl_resource_get_user_data(resource);
+	if ((surface->pending.invalid & WLR_SURFACE_INVALID_OPAQUE_REGION)) {
+		pixman_region32_clear(&surface->pending.opaque);
```
Comment (SirCmpwn) [inline types/wlr_surface.c]: Is this correct? Other regions accumulate according to the protocol.

File: types/wlr_surface.c [hunk 3]
```diff
@@ -199,46 +204,69 @@ static void wlr_surface_to_buffer_region(struct wlr_surface *surface,
...
+	int update_size = 0;
+	int update_damage = 0;
 
+	if ((surface->pending.invalid & WLR_SURFACE_INVALID_SCALE)) {
+		surface->current.scale = surface->pending.scale;
+		update_size = 1;
+	}
+	if ((surface->pending.invalid & WLR_SURFACE_INVALID_TRANSFORM)) {
+		surface->current.transform = surface->pending.transform;
+		update_size = 1;
```
Comment (SirCmpwn) [inline types/wlr_surface.c]: Make these stdbool.h booleans

File: backend/wayland/output.c (hunk 1 above)
Comment (ascent12) [inline backend/wayland/output.c]: wlr_output_init sets it already/

Comment (martinetd) [inline backend/wayland/output.c]: Yeah, it's just a follow-up on that earlier fix from yesterday. We no longer need it here.

File: types/wlr_surface.c (hunk 2 above)
Comment (martinetd) [inline types/wlr_surface.c]: It's what we do below anyway (we just copy over, while the other two damage callback do unions)
The clear only frees memory if that is called twice. I'll check what the protocol wants us to do.

Comment (martinetd) [inline types/wlr_surface.c]: I think it's fine to copy.
wayland-server-protocol.h says this for damage, which is clearly union:
```
         * The initial value for pending damage is empty: no damage.
         * wl_surface.damage adds pending damage: the new pending damage is
         * the union of old pending damage and the given rectangle.
```

On the other hand, opaque and input have nothing about accumulating, instead they just say 'copy'
For opaque
```
         * The initial value for an opaque region is empty. Setting the
         * pending opaque region has copy semantics [...]
```
And input
```
         * The initial value for an input region is infinite. That means
         * the whole surface will accept input. Setting the pending input
         * region has copy semantics...
```

File: types/wlr_surface.c (hunk 3 above)
Comment (martinetd) [inline types/wlr_surface.c]: Done!

**Comment by SirCmpwn** [comment]
Thanks!
